app/lib/provider_discovery.rb

```diff
@@ -22,6 +22,7 @@ def discover_provider(url, options = {})
         format ||= :xml if provider_endpoint
       end
 
+      raise OEmbed::NotFound, url unless provider_endpoint
```

Comment:
`Addressable::URI.parse` returns `nil` only when given uri is `nil`.  https://github.com/sporkmonger/addressable/blob/master/lib/addressable/uri.rb#L87

In other case, it always returns instance of it.

so there is other way like this:
```ruby
begin
   provider_endpoint = Addressable::URI.parse(provider_endpoint)
   raise OEmbed::NotFound, url unless provider_endpoint
   provider_endpoint.query = nil
   provider_endpoint = provider_endpoint.to_s
rescue Addressable::URI::InvalidURIError
   raise OEmbed::NotFound, url
end
```